### PR TITLE
chore(docs): translate all the code samples

### LINKED ISSUE #1929: Add multi-language code examples to docs
Most of our docs only include TypeScript examples. We should add multi-language code blocks to examples in our documentation to highlight that there are five supported languages to use with CDKTF.

Scanning the codebase shows that there are currently 26 code blocks in our docs, so this work could be handled incrementally. 

Reference: [Guidance from the Education team on using CodeTabs to add multiple languages to code examples](https://docs.google.com/document/d/1fmpSviM2pQUWm6y3QnVQ3QAdq5i7RjeWopzNXbMH2Io/edit#heading=h.sjgr1uoclelk)

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Closes #1929

### CODE REVIEW

File: website/docs/cdktf/concepts/constructs.mdx [hunk 1]
```diff
@@ -87,12 +87,15 @@ You can instantiate a construct multiple times throughout your infrastructure. F
 
 The example below creates three different S3 buckets, two of which are in the same scope. When CDKTF synthesizes this configuration, the Terraform IDs for these resources will be the construct names prefixed by the stack name and suffixed with a hash for each Construct instance.
 
+<CodeTabs>
 ```ts
```
Comment (BRKalow) [inline website/docs/cdktf/concepts/constructs.mdx:91]:
```suggestion
<CodeTabs>

```ts
```

File: website/docs/cdktf/concepts/constructs.mdx [hunk 2]
```diff
@@ -135,3 +426,7 @@ You can search the Amazon Web Services (AWS) [Construct Hub](https://constructs.
 If you want to distribute construct libraries, we recommend writing the constructs in TypeScript and using `jsii` to generate native packages for all desired language targets. We recommend that open source projects aim to support all generally available languages.
 
 You can use the [package generator projen](https://github.com/projen/projen#getting-started) to scaffold a Github project with the necessary Github Actions workflows to build and publish your construct in multiple programming languages.
+
+```
+
+```
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:432]: intentional?

File: website/docs/cdktf/create-and-deploy/best-practices.mdx [hunk 3]
```diff
@@ -55,46 +182,224 @@ You do not always need to create a new class for each stack. For some use cases,
...
+App app = new App();
+NetworkingStack devNetworking = new NetworkingStack(app, "networking-development", new Dictionary<string, object> { });
+
+// A single region is enough to create a development environment for each product.evelopment environment for each product.
+new Ecommerce(app, "ecommerce-development", new EcommerceProps {
+    VpcId = devNetworking.VpcId,
+    Subnets = devNetworking.Subnets,
+    Region = "us-west-1"
+});
+new Blog(app, "ecommerce-development", new BlogProps {
```
Comment (ansgarm) [inline website/docs/cdktf/create-and-deploy/best-practices.mdx:330]: Those two constructs have the same name which should fail, shouldn't it?
I think another Blog construct further down is also called "ecommerce" in its it as well.

Also true for other languages.

**Review by ansgarm — CHANGES_REQUESTED** [review]

File: website/docs/cdktf/concepts/assets.mdx [hunk 4]
```diff
@@ -60,6 +60,160 @@ new MyStack(app, "demo");
...
+
+	aws.NewAwsProvider(this, jsii.String("provider"), &awsProviderConfig{
+		region: jsii.String("us-west-2"),
+	})
+
+	bucket := aws.S3.NewS3Bucket(this, jsii.String("bucket"), &aws.s3.s3BucketConfig{
+		bucket: jsii.String("demo"),
+	})
+
+	asset := cdktf.NewTerraformAsset(this, jsii.String("lambda-asset"), &terraformAssetConfig{
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/assets.mdx:197]: This also needs to be `&cdktf.TerraformAssetConfig{`

Maybe testing the snippets will find more errors as I'm a bad Go compiler and there could be more missing.

File: website/docs/cdktf/concepts/assets.mdx [hunk 5]
```diff
@@ -60,6 +60,160 @@ new MyStack(app, "demo");
...
+		bucket: bucket.bucket,
+		key: asset.fileName,
+		source: asset.path,
+	})
+	return this
+}
+
+app := cdktf.NewApp()
+NewMyStack(app, jsii.String("demo"))
+app.synth()
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/assets.mdx:212]:
```suggestion
app.Synth()
```

File: website/docs/cdktf/concepts/constructs.mdx [hunk 6]
```diff
@@ -81,18 +83,157 @@ new MyStack(app, "app");
...
+		"app": jsii.String("myapp"),
+		"component": jsii.String("frontend"),
+		"environment": jsii.String("dev"),
+	})
+	return this
+}
+
+app := cdktf.NewApp()
+NewMyStack(app, jsii.String("app"))
+app.synth()
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:216]: also needs a capital `S`

File: website/docs/cdktf/concepts/constructs.mdx [hunk 7]
```diff
@@ -120,6 +261,156 @@ class MyStack extends TerraformStack {
...
+```go
+import "github.com/aws/constructs-go/constructs"
+import "github.com/hashicorp/terraform-cdk-go/cdktf"
+import "github.com/myorg/myproject/myconstructs"
+
+type publicS3Bucket struct {
+	construct
+	bucket s3Bucket
+}
+func newPublicS3Bucket(scope construct, name *string) *publicS3Bucket {
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:376]:
```suggestion
func NewPublicS3Bucket(scope construct, name *string) *publicS3Bucket {
```

File: website/docs/cdktf/concepts/constructs.mdx [hunk 8]
```diff
@@ -120,6 +261,156 @@ class MyStack extends TerraformStack {
...
+	return this
+}
+
+type MyStack struct {
+	terraformStack
+}
+
+func NewMyStack(scope construct, name *string) *MyStack {
+	this := &MyStack{}
+	newTerraformStack_Override(this, scope, name)
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:400]:
```suggestion
	NewTerraformStack_Override(this, scope, name)
```

File: website/docs/cdktf/concepts/data-sources.mdx [hunk 9]
```diff
@@ -21,46 +21,224 @@ Data Sources are part of a [Terraform provider](/cdktf/concepts/providers-and-re
...
+type HelloTerraform struct {
+	terraformStack
+}
+
+func NewHelloTerraform(scope construct, id *string) *HelloTerraform {
+	this := &HelloTerraform{}
+	newTerraformStack_Override(this, scope, id)
+
+	region := aws.Datasources.NewDataAwsRegion(this, jsii.String("region"))
+	cdktf.NewTerraformOutput(this, jsii.String("region"), &terraformOutputConfig{
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/data-sources.mdx:108]:
```suggestion
	cdktf.NewTerraformOutput(this, jsii.String("region"), &cdktf.TerraformOutputConfig{
```

**Review by ansgarm — CHANGES_REQUESTED** [review]
I think we should test the Go code that's in the examples as it is still containing some errors

File: website/docs/cdktf/concepts/constructs.mdx [hunk 10]
```diff
@@ -81,18 +83,157 @@ new MyStack(app, "app");
...
+}
+
+func NewMyStack(scope construct, name *string) *myStack {
+	this := &myStack{}
+	NewTerraformStack_Override(this, scope, name)
+
+  cwd, err := os.Getwd()
+  if err != nil {
+      log.Println(err)
+  }
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:198]: indentation seems off

File: website/docs/cdktf/concepts/constructs.mdx [hunk 11]
```diff
@@ -120,6 +261,156 @@ class MyStack extends TerraformStack {
...
+}
+```
+
+```go
+import "github.com/aws/constructs-go/constructs"
+import "github.com/hashicorp/terraform-cdk-go/cdktf"
+import "github.com/myorg/myproject/myconstructs"
+
+type publicS3Bucket struct {
+	construct
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:373]: I think this needs a type
```suggestion
	construct constructs.Construct
```

File: website/docs/cdktf/concepts/constructs.mdx [hunk 12]
```diff
@@ -120,6 +261,156 @@ class MyStack extends TerraformStack {
...
+	bucket s3Bucket
```
Comment (ansgarm) [inline website/docs/cdktf/concepts/constructs.mdx:374]: requires the namespace I think, possibly needs to be `myconstructs.S3Bucket`?

**Review by jsteinich — CHANGES_REQUESTED** [review]
Just scanning through I can see a handful of issues:

- Some current directory lookup errors
- `Map.of`/`List.of` Java issues

There are probably other issues as jsii isn't perfect at translating. Most sure why to find them all would be to synth each example and compare.

**Pull request closed without merging**